pkg/controller/elasticsearch/remotecluster/remoteca/controller.go

```diff
@@ -0,0 +1,474 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
Generally I'm a bit lost in what's remote and what's local in this entire file.
I think that's because I initially misunderstood what we want to do here.
I thought since we only deal with unidirectionnal remote clusters we only had to copy the remote cluster CA into the local cluster namespace. But IIUC even though the remote cluster is unidirectionnal from ES point of view, our certificate handling needs to be bidirectionnal?

I don't know if we can arrange the code in such a way that this is a bit more explicit.

Reply:
Even if the connection is unidirectional both certificate authorities must be copied.  So, Yes, the local cluster must become the remote one at some point.

I'll try to add more comments.